### PR TITLE
Add ability to notify WAITING_APPROVAL event to Slack

### PR DESCRIPTION
**Pull request by ono-max**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes part of #2342

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/notifier/slack.go [hunk 1]
```diff
@@ -170,6 +170,16 @@ func (s *slack) buildSlackMessage(event model.NotificationEvent, webURL string)
...
 		generateDeploymentEventData(md.Deployment, md.EnvName)
 
+	case model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL:
+		md := event.Metadata.(*model.NotificationEventDeploymentWaitApproval)
+		title = fmt.Sprintf("Deployment for %q is waiting approve", md.Deployment.ApplicationName)
+		approvers := ""
+		for _, v := range md.Approver {
+			approvers += " <@" + v + ">"
+		}
+		text = fmt.Sprintf("Approval is required from %s", approvers)
```
Comment (nghialv) [inline pkg/app/piped/notifier/slack.go:180]:
```suggestion
		approvers := make([]string, 0, len(md.Approvers)
		for _, a := range md.Approvers {
			approvers = append(approvers, fmt.Sprintf("<@%s>", a))
		}
		text = fmt.Sprintf("Approvable users: %s", strings.Join(approvers, " "))
```

File: pkg/model/notificationevent.proto [hunk 2]
```diff
@@ -88,6 +89,12 @@ message NotificationEventDeploymentCancelled {
     string commander = 3;
 }
 
+message NotificationEventDeploymentWaitApproval {
+    Deployment deployment = 1 [(validate.rules).message.required = true];
+    string env_name = 2 [(validate.rules).string.min_len = 1];
+    repeated string approver = 3;
```
Comment (nghialv) [inline pkg/model/notificationevent.proto:95]:
```suggestion
    repeated string approvers = 3;
```

File: pkg/app/piped/notifier/slack.go [hunk 3]
```diff
@@ -170,6 +170,16 @@ func (s *slack) buildSlackMessage(event model.NotificationEvent, webURL string)
 		text = md.Summary
 		generateDeploymentEventData(md.Deployment, md.EnvName)
 
+	case model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL:
+		md := event.Metadata.(*model.NotificationEventDeploymentWaitApproval)
+		title = fmt.Sprintf("Deployment for %q is waiting approve", md.Deployment.ApplicationName)
```
Comment (nghialv) [inline pkg/app/piped/notifier/slack.go:175]:
```suggestion
		title = fmt.Sprintf("Deployment for %q is waiting for an approval", md.Deployment.ApplicationName)
```

File: pkg/app/piped/notifier/slack.go (hunk 1 above)
Comment (ono-max) [reply pkg/app/piped/notifier/slack.go:180]: @nghialv 

Thanks! Maybe, `if len(md.Approvers)`? The variable `approvers` is not declared yet at 176th line.

Comment (nghialv) [reply pkg/app/piped/notifier/slack.go:180]: You are right. It should be that.

**Comment by nghialv** [comment]
Hmm. I think we are having some problems with our Kapetanios system.
Could you please re-format the file manually from your local?

**Comment by ono-max** [comment]
@nghialv 

I've updated it 👍

**Comment by nghialv** [comment]
Nice. Thank you.
/lgtm

**Comment by khanhtc1202** [comment]
Nice
/approve

**Pull request merged**